From a3696683b1d85cbdd4f433c3be8d0fdfd234f121 Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Fri, 28 Mar 2025 18:55:39 +0800
Subject: [PATCH 2/2] Fix string matching for AppleClang

Signed-off-by: nyanmisaka <nst799610810@gmail.com>
---
 source/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 0bcbddb18..d69b821af 100755
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -168,7 +168,7 @@ if(APPLE)
   add_definitions(-DMACOS=1)
 endif()
 
-if(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+if(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
     set(CLANG 1)
 endif()
 if(${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
-- 
2.34.1

